### ISSUE TITLE
supplying variable names for one cross-section only

### ISSUE THREAD

**Issue by edoardociscato** [floswald/psidR#17]
Hi,

I found this code very useful but thought of a possible improvement. Supplying `fam` and `ind` data.tables as in your example requires manually copy-pasting and formatting the list of variables somewhere (you seem to have some txt files with a variable database in the example) and one has to look up each variable on psidonline: if many variables are needed, this takes time and it is definitely very boring.

Instead of providing the variable original names for each wave of the survey, it would be convenient to be allowed to provide the variable original name for one wave only in which the variable is present, and let R look up for the original names of the variable in the waves where it is available. In this way, I can look, say, at the dictionaries for the 1970 wave, choose my variables straight from the documentation and without having to look them up, and then run this function to (1) find out in which waves the variables are actually available (2) have read-to-use inputs for your `build.panel` function.

I wrote a function to do so based on some very primitive lines of web scraping (using XML package). I cannot guarantee that this works perfectly for any variable, but for now I had no problem. This is how it looks:

```
years=c(1968:1997, seq(from=1999, to=2011, by=2))
i <- c("ER30046", "ER30052", "ER30066") # Variable original names for the 1970 wave
f <- c("V1567", "V1514") # Variable original names for the 1970 wave
ind <- get_variable_names(i, years)
fam <- get_variable_names(f, years)

> head(fam)
     year heads_avg_hrly_ern tot_fu_mon_inc_ov414
1968 1968               V337                  V81
1969 1969               V871                 V529
1970 1970              V1567                V1514
1971 1971              V2279                V2226
1972 1972              V2906                V2852
1973 1973              V3275                V3256

> head(ind)
     year age_of_individual grade_finished individual_weight
1968 1968           ER30004        ER30010           ER30019
1969 1969           ER30023             NA           ER30042
1970 1970           ER30046        ER30052           ER30066
1971 1971           ER30070        ER30076           ER30090
1972 1972           ER30094        ER30100           ER30116
1973 1973           ER30120        ER30126           ER30137
```

I also let the code pick the names of the variables using the denomination used on psidonline. Of course one could always change them later. Also, note that `grade_finished` (corresponding to `educ` in your code) is actually available in 1968 but missing in 1969, while `individual_weight` is not available after 1992. After checking on psidonline, I confirm that there is no mistake. However, the code does not let you choose whether to exclude `grade_finished` in 1968, if you have a reason to do so. This change should be performed later when the outcome panel is obtained.

What do you think about it? Would you be interested in having the function added to your package or integrated within `build.panel`?

**Comment by floswald** [comment]
yes, sounds great!
can you file a pull request on this repo with your new function? please add
a test along the lines of what yu just said above (check that variable
names correspond correclty, and that NAs show up in the right place).

i can explain if this is your first pull request, let me know.

**Comment by edoardociscato** [comment]
Sure, so I will run a few tests to check everything works fine. There might still be a very small probability that my simple web scraping strategy does not work with all variables. I'll file the pull request soon! That would actually be the first one outside of your class, so I'll let you know if I get into trouble.

**Comment by floswald** [comment]
ok - the only thing you need to do is `fork` this repo (button top right of this page). then you need to `git push` your changes onto your fork. (you cannot push directly into this [my] repo). get in touch with problems.

**Comment by floswald** [comment]
closed via #18.
